src/Form/Extension/Product/ProductTranslationTypeExtension.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Setono\SyliusRedirectPlugin\Form\Extension\Product;
+
+use Sylius\Bundle\ProductBundle\Form\Type\ProductTranslationType;
+use Sylius\Component\Core\Model\ProductTranslationInterface;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+final class ProductTranslationTypeExtension extends AbstractTypeExtension
+{
+    public static function getExtendedTypes(): array
+    {
+        return [ProductTranslationType::class];
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event): void {
+            $form = $event->getForm();
+            $data = $event->getData();
+
+            if (!$data instanceof ProductTranslationInterface || $data->getId() === null) {
+                return;
+            }
+
+            // Rework the form to put the new field just after Slug one
```

Comment:
You should update the associated twig template to put the checkbox below the slug :)

Reply:
There is no associated template. The form is rendered by `{{ translationFormWithSlug(form.translations, '@SyliusAdmin/Product/_slugField.html.twig', product) }}`
Hence the difficulty to customize it.

But indeed, I'm not very proud of this method.